### PR TITLE
EC-102] Add TBR fields

### PR DESCRIPTION
**Pull request by gcleary**
also add Auth error code
also add comments to document debug fields

### CODE REVIEW

File: el.proto [hunk 1]
```diff
@@ -70,11 +71,20 @@ message Rq {
...
+    bytes ctx = 15; // workspace memory block
+    bytes cache = 16; // cache contents memory block
+    TbrInfo tbr = 17; // TBR fields
+}
+
+message TbrInfo {
+    int32 token_id = 1;  // 0 means that TBR is not being used in this request
+    bytes device_id = 2; // device_id is populated here for TBR registration, and not in Rq
+    bytes sku = 3;       // SKU or model must be populated for TBR registration
+    int32 cc = 4;        // country code is optional
```
Comment (tboinske-skyhook) [inline el.proto:87]: Would it make sense to rename this message to `TbrRegistration`, and move `token_id` into the top level of `Message Rq` and also into `Message Rs`? `token_id` would be present in all location requests iff TBR is enabled for that client. `TbrRegistration` would be included only for registration requests. `token_id` would be present in a response message only in the case that the response is a registration response message.

File: el.proto [hunk 2]
```diff
@@ -70,11 +71,20 @@ message Rq {
     NbiotCells nbiot_cells = 8 [deprecated=true];
     Gnss gnss = 9;
     repeated Cell cells = 10;
-    uint32 sequence = 12;
+    // 11 vaps above
```
[the remark this answers is not in the document]
Comment (gcleary) [reply el.proto]: Yup, done.

File: el.proto (hunk 1 above)
Comment (gcleary) [reply el.proto]: Yup, done.

File: el.proto [hunk 3]
```diff
@@ -63,18 +71,19 @@ message Rq {
...
+    uint32 threshold = 14; // score is compared to this
+    bytes ctx = 15; // workspace memory block
+    bytes cache = 16; // cache contents memory block
+    int32 token_id = 17;  // 0 means that TBR is not being used in this request
+    TbrRegistration tbr = 18; // TBR registration fields
+}
+
+message TbrRegistration {
+    bytes device_id = 2; // device_id is populated here for TBR registration, and not in Rq
+    bytes sku = 3;       // SKU or model must be populated for TBR registration
```
Comment (tboinske-skyhook) [inline el.proto:85]: Not sure that the comments here in these two lines are necessary given that this message is sent _only_ in the case of TBR registration.

File: el.proto [hunk 4]
```diff
@@ -195,6 +204,7 @@ message Rs {
                                       // Least Sig Bit of last byte corresponds to first AP
                                       // Least Sig Bit of second to last byte corresponds to ninth AP etc
                                       // Aps are counted based on their position in the request
+    int32 token_id = 7;               // non-0 means that TBR registration was successful
```
Comment (tboinske-skyhook) [inline el.proto:207]:
```
// present (i.e., non-zero) only for successful registration response messages.
```

File: el.proto [hunk 5]
```diff
@@ -63,18 +71,19 @@ message Rq {
...
-    uint32 sequence = 12;
+    uint32 sequence = 12; // number of client requests processed
     uint32 score = 13; // this could be a repeated array with values for all cachelines
-    uint32 threshold = 14;
-    bytes ctx = 15;
-    bytes cache = 16;
+    uint32 threshold = 14; // score is compared to this
+    bytes ctx = 15; // workspace memory block
+    bytes cache = 16; // cache contents memory block
+    int32 token_id = 17;  // 0 means that TBR is not being used in this request
```
Comment (tboinske-skyhook) [inline el.proto:79]:
```
// present (i.e., non-zero) only for clients using TBR.
```

File: el.proto [hunk 6]
```diff
@@ -63,18 +71,19 @@ message Rq {
...
-    NbiotCells nbiot_cells = 8 [deprecated=true];
-    Gnss gnss = 9;
-    repeated Cell cells = 10;
-    uint32 sequence = 12;
+    uint32 sequence = 12; // number of client requests processed
     uint32 score = 13; // this could be a repeated array with values for all cachelines
-    uint32 threshold = 14;
-    bytes ctx = 15;
-    bytes cache = 16;
+    uint32 threshold = 14; // score is compared to this
```
Comment (tboinske-skyhook) [inline el.proto:76]: Can we rename this to `score_threshold`?

File: el.proto [hunk 7]
```diff
@@ -63,18 +71,19 @@ message Rq {
...
     bytes vaps = 11;
-    GsmCells gsm_cells = 4 [deprecated=true];
-    UmtsCells umts_cells = 5 [deprecated=true];
-    LteCells lte_cells = 6 [deprecated=true];
-    CdmaCells cdma_cells = 7 [deprecated=true];
-    NbiotCells nbiot_cells = 8 [deprecated=true];
-    Gnss gnss = 9;
-    repeated Cell cells = 10;
-    uint32 sequence = 12;
+    uint32 sequence = 12; // number of client requests processed
```
Comment (tboinske-skyhook) [inline el.proto:74]: Rename to `seq_num`?

**Review by tboinske-skyhook — APPROVED** [review]
Just some comment and field renaming suggestions. Apply these if you agree they represent improvements, and merge at will.

File: el.proto (hunk 3 above)
Comment (gcleary) [reply el.proto:85]: yup

File: el.proto (hunk 4 above)
Comment (gcleary) [reply el.proto:207]: yup

File: el.proto (hunk 6 above)
Comment (gcleary) [reply el.proto:76]: yup

File: el.proto (hunk 7 above)
Comment (gcleary) [reply el.proto:74]: yup

File: el.proto (hunk 5 above)
Comment (gcleary) [reply el.proto:79]: yup

**Pull request merged**